### PR TITLE
Adding waypoint support for Driving Route mode

### PR DESCRIPTION
**Pull request by brucearmstrong**
- Added "Use Wifi Only" setting. If value is false, carrierName comes from prefs. If true, carrierName comes from me.getCarrierName(), which currently is "" for the wifi-only case.
- Remove hard-coded 8008 from Cloudlet.java latency test code, and EVENT_LATENCY_REQUEST handling code. Use port info returned by the DME.
- Added CSS to "About" HTML to the use MobiledgeX colors instead of stark black and white.
- To add a waypoint, after driving route created, long-press on map, and select "Add waypoint to route".
- Individual waypoints can be dragged and dropped just like the start and end markers.
- New defaults/max values for animation times.
- Bug fixes when switching between driving and flying route modes.

### CODE REVIEW

**Review by lgarner — APPROVED** [review]
LGTM.

File: android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MainActivity.java [hunk 1]
```diff
@@ -1092,28 +1110,39 @@ public void onClick(DialogInterface dialog, int which) {
...
-                    default:
-                        Log.i(TAG, "Unknown dialog selection.");
+
+                String selectedItemText = items.get(which);
+
+                if (selectedItemText.equals(spoofText)) {
+                    Log.i(TAG, "Spoofing GPS at " + location);
+                    showMessage("GPS spoofing activated.");
+                    float[] results = new float[1];
+                    Location.distanceBetween(oldLatLng.latitude, oldLatLng.longitude, spoofLatLng.latitude, spoofLatLng.longitude, results);
```
Comment (franklin-huang-mobiledgex) [inline android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MainActivity.java:1120]: is this a built in function? Seems odd to return the distance between 2 points in an array

**Review by franklin-huang-mobiledgex — APPROVED** [review]

File: android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MainActivity.java (hunk 1 above)
Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MainActivity.java:1120]: Yes, it's built in. https://developer.android.com/reference/android/location/Location.html#distanceBetween(double,%20double,%20double,%20double,%20float[])

Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MainActivity.java:1120]: The other array entries are for bearings.

**Pull request merged**